lib/types/local-date.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+var utils = require('../utils');
+/** @module types */
+
+/** @const */
+var millisecondsPerDay = 86400000;
+/** @const */
+var dateCenter = Math.pow(2,31);
+/**
+ *
+ * Creates a new instance of LocalDate.
+ * @class
+ * @classdesc A date without a time-zone in the ISO-8601 calendar system, such as 2010-08-05.
+ * <p>
+ *   LocalDate is an immutable object that represents a date, often viewed as year-month-day. For example, the value "1st October 2014" can be stored in a LocalDate.
+ * </p>
+ * <p>
+ *   This class does not store or represent a time or time-zone. Instead, it is a description of the date, as used for birthdays. It cannot represent an instant on the time-line without additional information such as an offset or time-zone.
+ * </p>
+ * @param {Number} year The year.
+ * @param {Number} month Between 1 and 12 inclusive.
+ * @param {Number} day Between 1 and the number of days in the given month of the given year.
+ * @constructor
+ */
+function LocalDate(year, month, day) {
+  //implementation detail: internally uses a UTC based date
+  if (typeof year === 'number' && typeof month === 'number' && typeof day === 'number') {
+    this.value = new Date(Date.UTC(year, month - 1, day));
+  }
+  else if (typeof month === 'undefined' && typeof day === 'undefined') {
+    if (typeof year === 'number') {
+      //in milliseconds since unix epoch
+      this.value = new Date(year);
+    }
+  }
+  if (!this.value) {
+    throw new Error('You must provide a valid year, month and day');
+  }
+  var self = this;
+  /**
+   * A number representing the year.
+   * @name year
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  /**
+   * A number between 1 and 12 inclusive representing the month.
+   * @name month
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  /**
+   * A number between 1 and the number of days in the given month of the given year (28, 29, 30, 31).
+   * @name day
+   * @type Number
+   * @memberof module:types~LocalDate#
+   */
+  Object.defineProperties(this, {
+    'year': { enumerable: true, get: function () {
+      return self.value.getUTCFullYear();
+    }},
+    'month': { enumerable: true, get: function () {
+      return self.value.getMonth() + 1;
+    }},
+    'day': { enumerable: true, get: function () {
+      return self.value.getDate();
+    }}
+  });
+}
+
+/**
+ * Creates a new instance of LocalDate using the current year, month and day from the system clock in the default time-zone.
+ */
+LocalDate.now = function () {
```

Comment:
Ah yes, now I'm in sync.  I was getting wrapped up in the internal implementation, but it turns out to not matter if you store it in UTC and look at it in UTC.   If you look at LocalDate.value with your local timezone, there's a chance it could appear off, but I think that is a consideration one needs to take into account when using value.